src/ebpf.rs

```diff
@@ -384,7 +387,27 @@ pub const BPF_CLS_MASK    : u8 = 0x07;
 pub const BPF_ALU_OP_MASK : u8 = 0xf0;
 
 /// Prototype of an eBPF helper function.
-pub type Helper = fn (u64, u64, u64, u64, u64) -> u64;
+pub type HelperFunction = fn (u64, u64, u64, u64, u64) -> u64;
+
+/// Prototype of an eBPF helper verification function.
+pub type HelperVerifier = fn (u64, u64, u64, u64, u64, &[&[u8]], &[&[u8]]) -> Result<(()), Error>;
+
+/// eBPF Helper pair
+/// 
+/// Includes both the helper function itself, but also an optional helper verification function
+/// that if present will be called first to validate the helper parameters.  A verification
+/// function is not needed if the helper treats its arguments as values but if one of
+/// the arguments represent a pointer then that pointer must be verified by the 
+/// verification function.
+/// 
+/// Note: native jitted programs do not have the ability to call the verification programs
```

Comment:
Is this something we can enforce by throwing an Error if the jit encounters a Helper with Some(HelperVerifier)?